networking_aci/plugins/ml2/drivers/mech_aci/cobra_client.py

```diff
@@ -37,6 +38,7 @@
                        rexc.HTTPError, LoginError)
 RETRY_EXCEPTIONS = FALLBACK_EXCEPTIONS + (SSLError, CommitError, QueryError)
 requests.packages.urllib3.disable_warnings()
+_TOKEN_REFRESH_LOCK = threading.Lock()
```

Comment:
Just a question out of interest, isn't cobra used from a greenthread, so python-thread locking doesn't work here?

Reply:
Yes, cobra is called from a greethread. I wasn't sure when I wrote this code if threading.Lock works with greenthreads, but concluded it should work, as it is used by oslo.concurrency, e.g. [here](https://github.com/openstack/oslo.concurrency/blob/master/oslo_concurrency/lockutils.py#L103).

I just checked in the aci driver backdoor and threading.Lock looks very monkeypatched :)
```
>>> threading.Lock.__module__
'eventlet.green.thread'
>>> threading.Lock.func_code
<code object allocate_lock at 0x7fe070972130, file "/var/lib/openstack/local/lib/python2.7/site-packages/eventlet/green/thread.py", line 78>

```

Reply:
what a mess ^^ Thanks for clarification, good to know